### PR TITLE
smaller base images

### PR DESCRIPTION
**Pull request by nadade**
- minimal debian image
- deriving from the minimal debian image rather than ubuntu:14.04 for nodejs-0.12 and nodejs-ruby images
- scripts to build, squash and push selected images to DockerHub directly (without using their built-in mechanism) - we will need to disable the automated DockerHub builds for these

### CODE REVIEW

**Comment by w-vi** [comment]
Maybe a stupid question but why we need minimal dev images? I think we should have minimal production images and start using them where appropriate, now we use `*-dev` even in more production focused envrionments. Well I actually think that the `-dev` and `production` doesn't really work for us ... we could happily use just `base` and be done with it.

**Comment by abtris** [comment]
@w-vi problem is with images about 1GB. It's look ECR have problem with correct download theses images.

**Comment by nadade** [comment]
@w-vi agreed that it does not make much sense, I just kept the naming consistent... I'd rename in a separate PR, though...

**Comment by w-vi** [comment]
Well images over 1GB are bloated that's true :)

@nadade Cool, I think it will make life easier

File: scripts/docker-build-images.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    PACKAGE_NAME="apiaryio/base-dev-$IMAGE_NAME"
+    echo "Building $PACKAGE_NAME based on $DOCKERFILE"
+    docker build -t $PACKAGE_NAME -f $DOCKERFILE $LOCATION
+    echo "Squashing $PACKAGE_NAME..."
+    docker save $PACKAGE_NAME > "/tmp/$IMAGE_NAME.tar"
+    sudo docker-squash -i "/tmp/$IMAGE_NAME.tar" -o "/tmp/$IMAGE_NAME-squashed.tar"
+    cat "/tmp/$IMAGE_NAME-squashed.tar" | docker load
+    echo "Squashed $PACKAGE_NAME"
+done
+echo "All done!"
```
Comment (abtris) [inline scripts/docker-build-images.sh]: nit: missing line end

File: scripts/docker-internal-release.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+for IMAGE_NAME in $IMAGES_TO_SQUASH
+do
+    echo "Preparing to push $IMAGE_NAME..."
+    PACKAGE_NAME="apiaryio/base-dev-$IMAGE_NAME"
+    echo "Pushing $PACKAGE_NAME..."
+    docker push $PACKAGE_NAME
+    echo "$PACKAGE_NAME pushed successfully"
+done
+echo "All done!"
```
Comment (abtris) [inline scripts/docker-internal-release.sh]: nit: missing line end

try set this in your IDE to add endline to end of file

File: scripts/install-dependencies.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+wget https://github.com/jwilder/docker-squash/releases/download/v0.2.0/docker-squash-linux-amd64-v0.2.0.tar.gz
+sudo tar -C /usr/local/bin -xzvf docker-squash-linux-amd64-v0.2.0.tar.gz
```
Comment (abtris) [inline scripts/install-dependencies.sh]: nit: line end

File: scripts/docker-internal-release.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    echo "No IMAGES_TO_SQUASH set, skipping the internal release"
+    exit 1
+fi
+
+echo "Authenticating with Docker Hub..."
+if [[ -z "$DOCKER_EMAIL" || -z "$DOCKER_USER" || -z "$DOCKER_PASS" ]]; then
+    echo "DockerHub credentials are missing (DOCKER_EMAIL, DOCKER_USER, DOCKER_PASS), cannot proceed."
+    exit 1
+fi
+docker login -e $DOCKER_EMAIL -u $DOCKER_USER -p $DOCKER_PASS
```
Comment (abtris) [inline scripts/docker-internal-release.sh]: missing some error management if `~/docker.cfg` isn't writable or login fail

**Comment by nadade** [comment]
@abtris comments addressed, can you take one more look?

**Pull request merged**